bot_modules/irc_commands.js

```diff
@@ -116,6 +116,32 @@ module.exports = {
 			bot.say(info.channel, `String "${title}" does not match entry %title%;`)
 		})
 	},
+	
+	/**
+	 *	entryid
+	 *
+	 */
+	entryid: (info, words) => {
+		let title = words.slice(1).join(' ')
+		let p
+		if (typeof title === 'undefined' || title.length < 2) {
+			p = botb_api.request(`entry/random`)
+		} else {
+			p = botb_api.request(`entry/load/${title}`)
+		}
+
+		p.then(data => {
+			if (data.length == 0) throw "No entry data returned!"
+
+			let entry
+			if (data.length == 1) entry = data[0]
+			if (data.length > 1) entry = data[Math.floor(Math.random() * data.length)]
```

Comment:
I don't think this will work at all; the API response can be an object or an array, and `length` is not a property of objects. Use `Array.isArray()` if you want to test whether you got an array.

Reply:
I'm not sure why `/entry/random` returns an array of 1 entry object, but `/entry/load/${id}` gives you simply an object. I probably wasn't paying attention when I built it. Sorry, thinking out loud. ;)
I don't think this is a good place to return a random entry as `!entry` without any arguments does that already. It'll also simple what you are trying to do.
```if (typeof data === 'object')``` would tell you if you got an object back from the api.
Also you can remove the `|| title.length < 2` because its either a defined integer or its a bad value.